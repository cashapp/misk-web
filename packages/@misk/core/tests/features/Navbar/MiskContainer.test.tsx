import React from "react"
import { cleanup, render } from "@testing-library/react"
import {
  MiskNavbarContainer,
  testAdminDashboardTabsUrl,
  testServiceMetadataUrl
} from "src/features/Navbar"

describe("Misk Navbar Container", () => {
  afterEach(cleanup)
  it("Container can render", () => {
    const { asFragment } = render(<MiskNavbarContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("Container renders with remote data", () => {
    const { asFragment } = render(
      <MiskNavbarContainer
        dashboardMetadataUrl={testAdminDashboardTabsUrl}
        serviceMetadataUrl={testServiceMetadataUrl}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
