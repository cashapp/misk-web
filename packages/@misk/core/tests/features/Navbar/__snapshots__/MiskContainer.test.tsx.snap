// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misk Navbar Container Container can render 1`] = `
<DocumentFragment>
  .emotion-10 {
  background-color: #29333a !important;
  padding-top: 10px !important;
  padding-bottom: 60px !important;
  position: fixed !important;
}

.emotion-2 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-2 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-2 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    width: 1170px;
  }
}

.emotion-1 {
  font-size: 13px !important;
  font-weight: 600 !important;
  position: relative;
  padding-top: 25px;
  padding-bottom: 27px;
}

@media (max-width:870px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

@media (min-width:992px) and (max-width:1085px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

@media (min-width:1200px) and (max-width:1285px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

.emotion-0 {
  font-size: 24px !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-right: 30px;
  color: #cecece;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
}

.emotion-0 a {
  color: #cecece !important;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a:hover {
  color: white !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-3 {
  color: #7a7e82 !important;
}

.emotion-3:hover {
  color: white;
}

.emotion-9 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-8 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-8 {
    padding: 50px 20px;
  }
}

.emotion-6 {
  display: inline-block;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-5 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
    class="bp3-navbar emotion-10"
  >
    <div
      class="emotion-2"
    >
      <div
        class="bp3-navbar-group bp3-align-left bp3-dark emotion-1"
      >
        <span
          class="emotion-0"
        >
          Misk
        </span>
      </div>
    </div>
    <div>
      <button
        class="bp3-button emotion-4"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="bp3-icon bp3-icon-menu emotion-3"
            icon="menu"
          >
            <svg
              data-icon="menu"
              height="32"
              viewBox="0 0 20 20"
              width="32"
            >
              <desc>
                menu
              </desc>
              <path
                d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
      </button>
      <div
        class="emotion-9"
      >
        <div
          class="bp3-collapse"
        >
          <div
            aria-hidden="true"
            class="bp3-collapse-body"
            style="transform: translateY(-0px);"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-2"
              >
                <div
                  class="emotion-6"
                >
                  <div
                    class="emotion-5"
                  >
                    <span>
                      <span />
                    </span>
                  </div>
                </div>
                <span />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
  .emotion-0 {
  position: relative;
  top: 110px;
  padding-left: 5px;
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    width: 1170px;
  }
}

<div
    class="emotion-0"
  >
    <div />
  </div>
</DocumentFragment>
`;

exports[`Misk Navbar Container Container renders with remote data 1`] = `
<DocumentFragment>
  .emotion-10 {
  background-color: #29333a !important;
  padding-top: 10px !important;
  padding-bottom: 60px !important;
  position: fixed !important;
}

.emotion-2 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-2 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-2 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    width: 1170px;
  }
}

.emotion-1 {
  font-size: 13px !important;
  font-weight: 600 !important;
  position: relative;
  padding-top: 25px;
  padding-bottom: 27px;
}

@media (max-width:870px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

@media (min-width:992px) and (max-width:1085px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

@media (min-width:1200px) and (max-width:1285px) {
  .emotion-1 {
    padding-left: 60px;
  }
}

.emotion-0 {
  font-size: 24px !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-right: 30px;
  color: #cecece;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
}

.emotion-0 a {
  color: #cecece !important;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a:hover {
  color: white !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  background-color: #29333a !important;
  box-shadow: none !important;
  background-image: none !important;
  top: 15px;
  left: 15px;
  position: absolute;
  z-index: 1020;
}

.emotion-3 {
  color: #7a7e82 !important;
}

.emotion-3:hover {
  color: white;
}

.emotion-9 {
  background-color: #29333a;
  display: block;
  margin: 60px -20px 0 -20px;
}

.emotion-8 {
  min-height: 250px;
  padding: 50px 0px;
  overflow-y: scroll;
  max-height: 100vh;
}

@media (max-width:768px) {
  .emotion-8 {
    padding: 50px 20px;
  }
}

.emotion-6 {
  display: inline-block;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-5 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
    class="bp3-navbar emotion-10"
  >
    <div
      class="emotion-2"
    >
      <div
        class="bp3-navbar-group bp3-align-left bp3-dark emotion-1"
      >
        <span
          class="emotion-0"
        >
          Misk
        </span>
      </div>
    </div>
    <div>
      <button
        class="bp3-button emotion-4"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="bp3-icon bp3-icon-menu emotion-3"
            icon="menu"
          >
            <svg
              data-icon="menu"
              height="32"
              viewBox="0 0 20 20"
              width="32"
            >
              <desc>
                menu
              </desc>
              <path
                d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
      </button>
      <div
        class="emotion-9"
      >
        <div
          class="bp3-collapse"
        >
          <div
            aria-hidden="true"
            class="bp3-collapse-body"
            style="transform: translateY(-0px);"
          >
            <div
              class="emotion-8"
            >
              <div
                class="emotion-2"
              >
                <div
                  class="emotion-6"
                >
                  <div
                    class="emotion-5"
                  >
                    <span>
                      <span />
                    </span>
                  </div>
                </div>
                <span />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
  .emotion-0 {
  position: relative;
  top: 110px;
  padding-left: 5px;
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    width: 1170px;
  }
}

<div
    class="emotion-0"
  >
    <div
      class="bp3-spinner"
    >
      <div
        class="bp3-spinner-animation"
      >
        <svg
          height="50"
          stroke-width="8.00"
          viewBox="1.00 1.00 98.00 98.00"
          width="50"
        >
          <path
            class="bp3-spinner-track"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          />
          <path
            class="bp3-spinner-head"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            pathLength="280"
            stroke-dasharray="280 280"
            stroke-dashoffset="210"
          />
        </svg>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
